packages/hadron-document/src/element.ts

```diff
@@ -836,6 +839,35 @@ export class Element extends EventEmitter {
     }
   }
 
+  getVisibleElements() {
+    if (!this.elements) {
+      return [];
+    }
+    return [...this.elements].slice(0, this.visibleElementsCount);
+  }
+
+  setVisibleElementsCount(newCount: number) {
+    if (!this._isExpandable(this.originalExpandableValue)) {
+      return;
+    }
+    this.visibleElementsCount = newCount;
+    this._bubbleUp(Events.VisibleElementsChanged, this, this.getRoot());
```

Comment:
This is not true if element is collapsed, right?

Reply:
ahh that's right - I missed that, thanks for spotting.

Reply:
addressed this in recent changes